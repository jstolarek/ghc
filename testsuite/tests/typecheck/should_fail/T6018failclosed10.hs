{-# LANGUAGE TypeFamilies, DataKinds, UndecidableInstances #-}

module T6018failclosed10 where

-- this one is a strange beast. Last equation is unreachable but it generates
-- information used by injectivity. So we accept it as a valid injective type
-- family but complain when the last equation is actually used
type family Bar a = r | r -> a where
    Bar Int  = Bool
    Bar Bool = Int
    Bar Bool = Char

bar :: Bar a -> Bar a
bar x = x

barapp2 :: Char
barapp2 = bar 'c'
