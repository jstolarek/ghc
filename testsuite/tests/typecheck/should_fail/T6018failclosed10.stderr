
T6018failclosed10.hs:17:11:
    Couldn't match type ‘Int’ with ‘Char’
    Expected type: Char
      Actual type: Bar Bool
    In the expression: bar 'c'
    In an equation for ‘barapp2’: barapp2 = bar 'c'

